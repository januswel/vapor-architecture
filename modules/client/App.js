import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import sum from 'vapor-domain'

export default () => (
  <View style={styles.container}>
    <Text style={styles.text}>{sum(1, 3)}</Text>
  </View>
)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  text: {
    fontSize: 32,
  },
})
