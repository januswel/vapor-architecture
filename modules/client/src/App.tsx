import * as React from 'react'
import { Provider } from 'react-redux'

import store from './store'
import Screen from './containers/Screen'

export default () => (
  <Provider store={store}>
    <Screen />
  </Provider>
)
