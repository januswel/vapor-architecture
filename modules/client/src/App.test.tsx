import * as React from 'react'
import * as renderer from 'react-test-renderer'

import App from './App'

describe('App', () => {
  it('renders correctly', () => {
    const component = renderer.create(<App />)
    expect(component).toMatchSnapshot()
  })
})
