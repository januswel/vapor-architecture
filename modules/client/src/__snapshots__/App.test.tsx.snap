// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "black",
        "color": "orange",
        "fontSize": 32,
        "margin": 8,
        "padding": 16,
        "textAlign": "right",
      }
    }
  >
    ¥ 
    0
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          10
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          50
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          100
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "silver",
            "borderRadius": 48,
            "height": 96,
            "justifyContent": "center",
            "width": 96,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          500
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 0,
          "imageUrl": "http://localhost:3000/images/coffee.png",
          "isSoldOut": false,
          "isTooFewChargedMoney": true,
          "price": 120,
        },
        Object {
          "id": 1,
          "imageUrl": "http://localhost:3000/images/water.png",
          "isSoldOut": false,
          "isTooFewChargedMoney": true,
          "price": 100,
        },
        Object {
          "id": 2,
          "imageUrl": "http://localhost:3000/images/soda.png",
          "isSoldOut": false,
          "isTooFewChargedMoney": true,
          "price": 120,
        },
        Object {
          "id": 3,
          "imageUrl": "http://localhost:3000/images/coffee.png",
          "isSoldOut": true,
          "isTooFewChargedMoney": true,
          "price": 120,
        },
      ]
    }
    disableVirtualization={false}
    extraData={0}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={2}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginTop": 16,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 168,
                  "justifyContent": "space-around",
                  "margin": 16,
                  "width": 168,
                },
                Object {
                  "backgroundColor": "gray",
                },
              ]
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "http://localhost:3000/images/coffee.png",
                }
              }
              style={
                Object {
                  "height": 96,
                  "width": 96,
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              ¥
              120
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 168,
                  "justifyContent": "space-around",
                  "margin": 16,
                  "width": 168,
                },
                Object {
                  "backgroundColor": "gray",
                },
              ]
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "http://localhost:3000/images/water.png",
                }
              }
              style={
                Object {
                  "height": 96,
                  "width": 96,
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              ¥
              100
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 168,
                  "justifyContent": "space-around",
                  "margin": 16,
                  "width": 168,
                },
                Object {
                  "backgroundColor": "gray",
                },
              ]
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "http://localhost:3000/images/soda.png",
                }
              }
              style={
                Object {
                  "height": 96,
                  "width": 96,
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              ¥
              120
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 168,
                  "justifyContent": "space-around",
                  "margin": 16,
                  "width": 168,
                },
                Object {
                  "backgroundColor": "red",
                },
              ]
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "http://localhost:3000/images/coffee.png",
                }
              }
              style={
                Object {
                  "height": 96,
                  "width": 96,
                }
              }
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              ¥
              120
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
